docs/building-extensions/plugins/plugin-events/content.md

```diff
@@ -0,0 +1,161 @@
+---
+sidebar_position: 1
+title: Content Plugin Events
+toc_max_heading_level: 2 
+---
+
+Content Plugin Events
+=====================
+
+Content events are triggered during the content creation process. These events are triggered in many different components and modules - they are not specific to the `com_content` component. 
+
+This list gives a brief description of each event, what the event parameters / arguments are, and any examples of their use which are available in the Joomla Manual.
+
+For background on Joomla transitioning to using classes for events see [Joomla 4 and 5 changes](../joomla-4-and-5-changes.md), where you can also find explanations for [accessing the arguments](../joomla-4-and-5-changes.md#summary---accessing-event-arguments) and [returning values](../joomla-4-and-5-changes.md#summary---returning-values). 
+
+## onContentPrepare
+
+### Description
+
+This is the first stage in preparing content for output and is the most common point for content-orientated plugins to do their work. 
+Since the item and related parameters are passed by reference, event handlers can modify them prior to display. 
+
+### Event Arguments
+The event class \Joomla\CMS\Event\Content\ContentPrepareEvent has the following arguments:
+
+- **`context`** - The context of the content being passed to the plugin. This is the component name and view - or name of module (e.g. com_content.article, com_contact.contact, com_users.user). Use this to check whether you are in the desired context for the plugin.
+
+- **`&subject`** (**`&item`** is also acceptable) - A reference to the item which is being rendered by the view, for example, an article, contact or user. 

```

Comment:
no no, sorry, this will not work :wink: 
What I meant, that `item` is a previous name of the parameter, and use of `item` does not work with event class anymore.
this will work: `$event->getArgument('subject')`;
this will return `null`:  `$event->getArgument('item')`;

Reply:
I checked and $event->getArgument('item') returns the item ok.

I'm finding this quite confusing though. For NormaliseRequestDataEvent you have

    $event->getArgument('context');
    $event->getArgument('data');
    $event->getArgument('subject');
    $event->getArgument('form');

And also 

    $event->getContext();
    $event->getData();
    $event->getForm();

but you can't use 

    $event->getSubject();

I think the guidance should be to use 

    $event->getContext();

instead of 

    $event->getArgument('context');

And then in the list of event arguments we should use the names which correspond to those functions which exist (ie use 'form' instead of 'subject').

There's a couple of other reasons for this:
- people will understand what 'form' means more readily than 'subject'
- it matches what's described in the API documents https://api.joomla.org/cms-5/classes/Joomla-CMS-Event-Model-NormaliseRequestDataEvent.html 

Do you foresee any problem with this approach?

Reply:
> I checked and $event->getArgument('item') returns the item ok.

hm, you are right, I just checked and this happen due to legacy code handling https://github.com/joomla/joomla-cms/blob/46c3bc031670aa0550ebbc7e738d73ab3c349782/libraries/src/Event/AbstractEvent.php#L170-L182
and will stop working at some point in future.
So, it works by accident :smile: 

> I'm finding this quite confusing ...

Yes, it also was for me. And It probably will be confusing for everyone :smile: 

It comes from https://github.com/joomla/joomla-cms/blob/46c3bc031670aa0550ebbc7e738d73ab3c349782/libraries/src/Event/AbstractEvent.php#L45-L46
And when I made event classes, I took the "main" parameter and name it as a `subject` argument. To follow that description.

Later I thought it much nicer to have the getters with typehint like `$event->getFoobar()` than use of `$event->getArgument('foobar')` or `$event['foobar']`. And added these getters, but without use of `getSubject()` for `subject` argument. 
We probably can add this method at some point, to reslove this confusion. Not sure currently.

 
> I think the guidance should be to use `$event->getContext();` instead of `$event->getArgument('context');`

Yes and no. If we talking about specific event, that have this method implemented, then yes. If we talking about generic event class, that does not have this method, then no.
I mean, these methods is kind of helpers, which provides type hinting.

Reply:
Yes, just for specific events. I think we focus on the getters in the documentation, for several reasons.

- 'subject' means different things in different events, so it's not as clear as 'form' or 'item'
- type-hinting available
- that's what the API docs show
- matches the "update" setters.

I'll update the documentation - again ;-)